### ISSUE TITLE
Hanging and crashing on a very specific export

### ISSUE THREAD

**Issue by thecsw** [darktable-org/darktable#18115]
### Describe the bug

I have an image and changes I've made to it. Same style applied to a whole collection of 20+ pictures works fine, but this one picture is completely not able to be exported out. Please find the .DNG and .XMP attached in a zip to this ticket as
[darktable_cannot_export.zip](https://github.com/user-attachments/files/18271353/darktable_cannot_export.zip)

The export settings I'm using are attached below,
<img width="273" alt="image" src="https://github.com/user-attachments/assets/c5270e31-da13-46ce-8ca8-9617ddc964ca" />

I can see that darktable is able to soft-proof and create a preview for the final image, but cannot get the fully processed picture out.

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/ce2a0081-c704-420a-9d21-2f301e563111" />

This appears to be the case for any export format. It would kind of sit in this state without telling any more nor failing.

<img width="271" alt="image" src="https://github.com/user-attachments/assets/a25deefe-13a8-4d2b-9ef6-bfdefa5fd5b9" />

As usual, I can see the spike in CPU usage, but then darktable just gets stuck and kind of idling without signs of doing really anything.

This seems to be consistent across machines. I'm running a MacBook Pro, 13-inch, M1, 2020 with 8GB RAM. 

I'm using the latest Darktable 5.0.0 for arm64 macOS.

Thank you! It's really odd to me, as darktable seems to be idling and minutes after, it still doesn't finish the export of this image, whereas every other image processes in about a second.

### Steps to reproduce

1. Download [darktable_cannot_export.zip](https://github.com/user-attachments/files/18271353/darktable_cannot_export.zip).
2. Try to export it as anything, for example
   - JPEG XL
   - bit depth: 16-bit
   - pixel type: unsigned integer
   - quality 90
   - encoding effort 5
   - decoding speed 2
   - allow upscaling
   - high quality resampling
3. Wait for it...

### Expected behavior

It should export within a second or two

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

5.0.0

### What OS are you using?

Mac

### What is the version of your OS?

MacOS 15.2

### Describe your system?

MacBook Pro, 13-inch, M1, 2020 with 8GB RAM. 

```
Darwin Mania.local 24.2.0 Darwin Kernel Version 24.2.0: Fri Dec  6 18:40:14 PST 2024; root:xnu-11215.61.5~2/RELEASE_ARM64_T8103 arm64
```

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

Using the "Apple" option

**Comment by thecsw** [comment]
I found out that "haze removal" triggers this odd behavior, where disabling it works fine. But some values work but some cause darktable to crash.

Setting it to strength 14 and distance 0.200 works fine, but say I change it to "strength +0.42 and distance 0.062," then it will hang indefinitely. 

I think at this point I'm okay with this as it's an issue with a singular photo on an array of settings for a single module—closing this and will dismiss as an anomaly. Thanks!

**Comment by ralfbrown** [comment]
It is in fact a bug for which the fix is current pending (#18113), but it only triggers, as you discovered, for specific combinations of settings and image.  The export would eventually have finished, but it might have taken days for that one image....

**Comment by thecsw** [comment]
@ralfbrown how interesting! I tried to build darktable from source, but some issues with mis-linked libraries and arch mismatches prevent me from doing it so quicky—thank you for catching this issue and finding a fix for it. What a coincidence that I hit a snag with it a few hours after you've opened the PR. 

One day I wish to understand the inner-workings of darktable better—it seems like a lot of fun to delve deeper into it. I'll try to see later again if I can figure out how to test out a tip or a branch to play with the same problematic settings.

Thank you and Happy New Year + Winter Holidays!
